consumer.go

```diff
@@ -44,18 +45,22 @@ consume:
 		iter := fetches.RecordIter()
 		for !iter.Done() {
 			record := iter.Next()
-			c.client.CommitRecords(ctx, record)
-			messages = append(messages, record.Value)
+			records = append(records, record)
 
-			if timeout.Err() != nil && len(messages) > 0 {
+			if timeout.Err() != nil && len(records) > 0 {
 				break consume
 			}
 
-			if len(messages) >= batchSize || len(errs) > 0 {
+			if len(records) >= batchSize || len(errs) > 0 {
 				break consume
 			}
 		}
 	}
+	c.client.CommitRecords(ctx, records...)
```

Comment:
Another option is to change this loop to be something like,
```go
for batchSize > 0 {
  fetches := cl.PollRecords(ctx, batchSize)
  fetches.EachError(...)
  iter := fetches.RecordIter()
  for !iter.Done() {
    r := iter.Next()
    messages = append(messages, record.Value)
    // more handling
  }
}
cl.CommitUncommittedRecords(ctx)
```

Reply:
If you do use `PollRecords`, I recommend updating the client to v1.1.1 -- I just fixed a bug that would manifest during cluster instability.

Reply:
thank you so much!